docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
+* Admin
+
+
+### Compute
+
+#### Nodes
+
+* EC2 compute nodes
+  * Application nodes
+  * Database nodes
+  * Bastion node
+* Auto-scaling groups
+
+
+#### Orchestration: Kubernetes
+
+#### Pods
+
+##### ReplicaSets
+
+* Traefik
+* Application
+* Database
+* cert-manager
+* Metrics server
+* bastion
+
+##### DaemonSets
+
+* Any ?
+
+
+### Storage
+
+#### CSI
+
+##### EBS
+
+* FIXME
+
+
+## CI/CD
+
+### GitHub Actions
+
+
+#### Actions
+
+#### Workflows
+
+* Application
+* Infrastructure
+
+
+## Observability
+
+### Log management (ELK/EFK)
+
+### Metrics (Prometheus/Grafana)
+
+### Event and alerting
+
+
+## Business continuity & Disaster Recovery
+
+### TODO Recap HA features
+
+### Backup
```

Comment:
Our image is immutable.
Our database is backed up automatically, using the built in feature of the postgres-operator.
There's a basebackup created every noon UTC, and the WAL is sent all through the day. The backup is stored in an S3 bucket in the US.

Reply:
Git backup: we looked into it, but have nothing ready at this point.
Artifacts backup: no plan at the moment.
Our recovery is 1 day by default, plus we have WAL transfers every 16 MB (I think that's the default). We're using default values, we can be more specific, if needed.
The backup is stored in an S3 bucket in the US.
The backup rotation is keeping the last 5 backups at the moment. We might want to fine-tune it.

Reply:
S3 looks like a very reliable backend, specially with CRR, and economically sound with Glacier transitioning.

It allows for implementation of 3/2/1 rule:
* 3 copies: easy one
* 2 medias: S3 + Glacier instant retrieval archive
* 1 "offsite": Glacier flexible/deep archive

As well as the good old GFFS:
* Son: daily to S3, rotated weekly/monthly
* Father: weekly to glacier instant, rotated monthly/quarterly
* Grandfather: monthly to glacier flexible, rotated bi-yearly/yearly

Oldest member of each generation could be transitioned to deep archive for an extended period of time. 

So, what about:

* GitHub backup to S3

  * https://github.com/marketplace/actions/s3-backup
  * https://rewind.com/blog/backing-up-your-github-repository-to-multiple-aws-s3-regions/
  * https://blog.marcolancini.it/2021/blog-github-backups-with-ecs/

* Deployment artifacts (container images): CRR and lifecycle policies 

  * https://disaster-recovery.workshop.aws/en/services/containers/ecr/ecr.html
  * https://aws.amazon.com/blogs/compute/clean-up-your-container-images-with-amazon-ecr-lifecycle-policies/
  * https://docs.aws.amazon.com/AmazonECR/latest/userguide/LifecyclePolicies.html

* EBS: one day looks like an eternity to me, could we add a pinch of snapshoting into that pot? Something like hourly